contracts/governance/TimelockController.sol

```diff
@@ -31,6 +31,47 @@ contract TimelockController is AccessControl, IERC721Receiver, IERC1155Receiver
     mapping(bytes32 => uint256) private _timestamps;
     uint256 private _minDelay;
 
+    enum OperationState {
+        Pending,
+        Ready,
+        Done
+    }
```

Comment:
I'd add this for operationId that are not yet scheduled. I think its a bad design that Governor.state reverts for these.
```suggestion
    enum OperationState {
        Unset,
        Pending,
        Ready,
        Done
    }
```

Reply:
Just added this, thanks for the idea.

> I think its a bad design that Governor.state reverts for these.

Would you mind clarifying this? Is this a suggestion of doing the same in `Governor`? I think it's definitely doable so we can get rid of the `GovernorInexistentProposal` and use `GovernorIncorrectState` instead.

Reply:
The governor "state()" is something we inherited from compond, and we can't change. Its beyond what we would be confortable changing (even in a major). Does that mean we can used a unset state in the errors ? maybe, I'd like to discuss that with fran because I'm not sure the implications would be.

Here we are free to do whatever we want, because there is no "state" enum.

Reply:
I see, thanks for clarifying!

I'd invoke @frangio using my special card

![image](https://github.com/OpenZeppelin/openzeppelin-contracts/assets/33379285/6036574f-6728-4686-bb4c-157e531b902c)

Is there something we can do in Governor to get rid of `GovernorInexistentProposal` and `GovernorDuplicatedProposal` by extending the Governor's enum?

Reply:
In theory we can add an `Unset` state but it would need to be added at the end of the enum to preserve the numeric values of other states, so the value of Unset would not be 0. Ideally, I think Unset should have numeric value 255, but there appears to be no way to do that in Solidity. Having Unset with numeric value 9 (the next one available currently) seems bad but I'm also not sure if there are concrete reasons to avoid it... The only I can think of is that other extensions could potentially extend the ProposalState enum in different ways and it would conflict.

It would be nice to simplify the errors. We could also use `type(uint8).max` as the "unset state" in the `IncorrectState` error. But this will not work if we define the argument as `ProposalState current`.

Not sure that these options are better than the current code in the PR.

Reply:
A "hacky" option would be to take the error very literally
```solidity
error GovernorIncorrectState(uint256 proposalId, ProposalState current, bytes32 expectedStates);
```

`expectedStates` is "bitmap" where each 1 represent an accepted state. We could use `bytes32(0)` to mean that no state is accepted / valid ... which would imply that we want the current proposal to not have a state. I'm not sure if it makes sens to you, but that interpretation would allow the use of this without tampering with the state enum.

Reply:
> A "hacky" option would be to take the error very literally
> 
> ```solidity
> error GovernorIncorrectState(uint256 proposalId, ProposalState current, bytes32 expectedStates);
> ```
> 
> `expectedStates` is "bitmap" where each 1 represent an accepted state. We could use `bytes32(0)` to mean that no state is accepted / valid ... which would imply that we want the current proposal to not have a state. I'm not sure if it makes sens to you, but that interpretation would allow the use of this without tampering with the state enum.

This option makes sense to me as long as we add it to the NatSpec of the error. Also, I think it makes sense because in Governor there's a set of expected states whereas in `TimelockIncorrectState` there's only a single expected state (0 is already taken by `Unset` and there's no other way of signaling the absence of a state).

I'd agree with doing this.

Reply:
That sounds fine to me, but I think it allows to represent the "duplicated" error and not the "inexistent" error. We can remove Duplicated and keep Inexistent as a separate revert.

Reply:
Removed the duplicated error in favor of the incorrect state. Kept the inexistent